### ISSUE TITLE
Accessing entity.body to grab while moving

### ISSUE THREAD

**Issue by bday336** [c-frame/aframe-extras#208]
I am trying to interact with an entity that is moving in the scene by a vector field I have input. I have been trying to use the grab.js script in the aframe-extras repository, however when I move the entity with my vector field component I can not grab it with the vive controller. I have attempted to have the entity be a dynamic body which will allow me to grab it, however when it is a dynamic object it will not move in the vector field as it did before. It will remain stationary. I had thought that the problem was that the body position of the dynamic body of the entity was not being updated so I tried printing out the body's position, however when I add a print to console for the entity.body.position it says that entity.body is undefined. Yet when I do the same print command in the browser console it shows the entity.body.position vector as expected. I have been trying different attempts to access the entity.body.position so I can update its position from my vector field motion.

In summary, I am generating an entity in the scene and giving it a component which moves it position through a given vector field. I want to be able to grab it while it is moving so that I can move it to a different position and release it so that it will begin to move due to the vector field again. I am currently not able to access the body component of the entity within the script.

This is my script where my vector field component is labeled spin1
[https://glitch.com/edit/#!/salt-work?path=views/index.html:101:35](url)

Could someone help me with accessing the body of the entity such that I can grab it with a vive controller and then release it back into the field?

**Comment by donmccurdy** [comment]
Hi, the link you include is not showing up for me, I just see a default Glitch page.

Have you considered using a constraint, as in the [`grab` component here](https://github.com/donmccurdy/aframe-extras/blob/master/src/misc/grab.js) and [this example](https://github.com/donmccurdy/aframe-extras/blob/master/examples/vive/index.html)? 

Dynamic bodies are meant to receive updates from the physics system primarily, but see this issue for some workarounds: https://github.com/donmccurdy/aframe-physics-system/issues/78

**Comment by bday336** [comment]
Sorry for the link, it was working fine for me when I posted. This one should work now
https://glitch.com/edit/#!/join/69f17c32-a409-409d-8c13-04a7608389a4

And I have been looking at some examples to try and model my controller interactions with what I want my script to accomplish. I have been looking at 

https://github.com/bryik/aframe-ball-throw/blob/master/index.html

which is what I have been basing my grab script off of. It seems similar to the example you mentioned, but I will look at your example as well.

When you say constraint are you meaning like the object identifier for the sphere-collider?

**Comment by donmccurdy** [comment]
> When you say constraint are you meaning like the object identifier for the sphere-collider?

Different than that, there's a number of `CANNON.Constraint` classes you can use. See [this snippet](https://github.com/donmccurdy/aframe-extras/blob/master/src/misc/grab.js#L70-L72) creating a constraint in JS, or [these examples ](https://github.com/donmccurdy/aframe-physics-system/blob/master/examples/constraints.html) creating constraints in HTML.

Also note there is an error in the JS console. `x.body is undefined`. The body won't be created until the component is fully initialized.

I can point you to examples but unfortunately can't help debug the vector field, may want to go to Stack Overflow for that, or even try using a simpler 2D JS physics engine.
